oauth2_http/java/com/google/auth/oauth2/CredentialAccessBoundary.java

```diff
@@ -167,7 +167,10 @@ public CredentialAccessBoundary build() {
           "The provided availableResource is either null or empty.");
       this.availableResource = availableResource;
 
-      this.availablePermissions = new ArrayList<>(checkNotNull(availablePermissions));
+      checkArgument(
+          availablePermissions != null && !availablePermissions.isEmpty(),
+          "The list of provided availablePermissions is either null or empty.");
```

Comment:
These should be two checks with two different messages. In general avoid messages like "Thing X is wrong or Thing Y is wrong" in favor of simply "Thing X is wrong"